### PR TITLE
Add selectedApp and isHelmManaged hooks to AppVersionHistoryRow

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- add app and isHelmManaged hook to AppVersionHistoryRow

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/68569/refactor-version-history-api-calls-to-use-hooks

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Does this PR require documentation?

none

### CODE REVIEW

File: web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx [hunk 1]
```diff
@@ -180,12 +186,12 @@ function AppVersionHistoryRow(props: Props) {
   };
 
   const renderVersionAction = (version: Version) => {
-    const app = props.app;
+    const app = selectedApp;
     const downstream = app?.downstream;
```
Comment (miaawong) [inline web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx:190]: Can this be destructured from selectedApp? 
```suggestion
const { downstream } = selectedApp
```

File: web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx [hunk 2]
```diff
@@ -300,7 +307,7 @@ function AppVersionHistoryRow(props: Props) {
               ) : preflightState.preflightState !== "" ? (
                 <>
                   <Link
-                    to={`/app/${app?.slug}/downstreams/${app?.downstream.cluster?.slug}/version-history/preflight/${version?.sequence}`}
+                    to={`/app/${app?.slug}/downstreams/${app?.downstream?.cluster?.slug}/version-history/preflight/${version?.sequence}`}
```
Comment (miaawong) [inline web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx:310]: could this be using selectedApp and downstream directly? 
```suggestion
                    to={`/app/${selectedApp?.slug}/downstreams/${downstream?.cluster?.slug}/version-history/preflight/${version?.sequence}`}
```

File: web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx [hunk 3]
```diff
@@ -180,12 +186,12 @@ function AppVersionHistoryRow(props: Props) {
   };
 
   const renderVersionAction = (version: Version) => {
-    const app = props.app;
+    const app = selectedApp;
```
Comment (miaawong) [inline web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx:189]: might not be necessary to reassign this to app

**Review by miaawong — APPROVED** [review]

File: web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx (hunk 3 above)
Comment (rianfowler) [reply web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx:189]: I’ll clean that up. Just wanted to keep the changes limited for now.

Comment (miaawong) [reply web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx:189]: yep no worries!  😊

**Pull request merged**